### ISSUE TITLE
if statement doesn't match implicit return

### ISSUE THREAD

**Issue by Songbird0** [rust-lang/rust#59995]
Hi all,

I tried this code:

```rust
fn foo() -> bool {
    if 1 == 1 {
        true
    }
    false
}
```

I expected to see this happen: no error about a potential mismatched type.

Instead, this happened: rustc doesn't recognize the implicit return instruction (i.e. `true`).

```rust
error[E0308]: mismatched types
 --> src/main.rs:3:9
  |
3 |         true
  |         ^^^^ expected (), found bool
  |
  = note: expected type `()`
             found type `bool`

error: aborting due to previous error

For more information about this error, try `rustc --explain E0308`.
error: Could not compile `playground`.

To learn more, run the command again with --verbose.
```

## Meta

rustc version: `1.34` (playground)

Is it a bug or if statement never recognized implicit return?

**Comment by jonas-schievink** [comment]
Only the last expression in the body can be implicitly returned (and recursively all of *that* expression's tail expressions). Closing as not a bug.

(questions like this are better asked on the user forum at https://users.rust-lang.org/)
